pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -584,13 +567,12 @@ func (ps *prometheusScraper) Report(socketFile string, vmi *k6tv1.VirtualMachine
 		}
 	}()
 
-	vmiMetrics := newVmiMetrics()
 	k8sLabels, k8sLabelValues := updateKubernetesLabels(vmi)
 
-	vmiMetrics.updateMemory(vmi, vmStats, ps.ch, k8sLabels, k8sLabelValues)
-	vmiMetrics.updateVcpu(vmi, vmStats, ps.ch, k8sLabels, k8sLabelValues)
-	vmiMetrics.updateBlock(vmi, vmStats, ps.ch, k8sLabels, k8sLabelValues)
-	vmiMetrics.updateNetwork(vmi, vmStats, ps.ch, k8sLabels, k8sLabelValues)
+	updateMemory(vmi, vmStats, ps.ch, k8sLabels, k8sLabelValues)
```

Comment:
Consider wrapping these four calls with one `updateMetrics` method on the struct which you then call here. It does not look like we want to call them independently as such. We want to have different scopes of concern, but the Reporter does not have to care about this, we can keep that internal to the struct.